src/Middleware/EventDispatcherMiddleware.php

```diff
@@ -44,26 +39,24 @@ public function __construct(EventDispatcherInterface $eventDispatcher, $clientId
      */
     public function push(HandlerStack $stack)
     {
-        $stack->push(function (callable $handler) {
-            return function (
-                RequestInterface $request,
-                array $options
-            ) use ($handler) {
-                $promise = $handler($request, $options);
+        $stack->push(fn (callable $handler) => function (
+            RequestInterface $request,
+            array $options
+        ) use ($handler) {
+            $promise = $handler($request, $options);
 
-                return $promise->then(
-                    function (ResponseInterface $response) use ($request) {
-                        $this->sendEvent($request, $response);
+            return $promise->then(
+                function (ResponseInterface $response) use ($request) {
+                    $this->sendEvent($request, $response);
 
-                        return $response;
-                    },
-                    function (\Exception $reason) use ($request) {
-                        $this->sendErrorEvent($request, $reason);
+                    return $response;
+                },
+                function (\Exception $reason) use ($request): never {
```

Comment:
I didn't know that we could define a return type on a closure... and I'm not sure to understand how it could be a good thing to do so... but I guess we can let rector do whatever it wants 🤷